src/zope/app/testing/xmlrpc.rst

```diff
@@ -0,0 +1,392 @@
+  >>> proxy.contents()
+  Traceback (most recent call last):
+  ...
+  zope.security.interfaces.Unauthorized: ...
+
+Parameters
+==========
+
+Of course, XML-RPC views can take parameters, too:
+
+  >>> class ParameterDemo:
+  ...     def __init__(self, context, request):
+  ...         self.context = context
+  ...         self.request = request
+  ...
+  ...     def add(self, first, second):
+  ...         return first + second
+
+Now we'll register it as a view:
+
+  >>> from zope.configuration import xmlconfig
+  >>> ignored = xmlconfig.string("""
+  ... <configure
+  ...     xmlns="http://namespaces.zope.org/zope"
+  ...     xmlns:xmlrpc="http://namespaces.zope.org/xmlrpc"
+  ...     >
+  ...   <!-- We only need to do this include in this example,
+  ...        Normally the include has already been done for us. -->
+  ...   <include package="zope.app.publisher.xmlrpc" file="meta.zcml" />
+  ...
+  ...   <xmlrpc:view
+  ...       for="zope.site.interfaces.IFolder"
+  ...       methods="add"
+  ...       class="zope.app.testing.xmlrpc.README.ParameterDemo"
+  ...       permission="zope.ManageContent"
+  ...       />
+  ... </configure>
+  ... """)
+
+Then we can issue a remote procedure call with a parameter and get
+back, surprise!, the sum:
+
+  >>> proxy = ServerProxy("http://mgr:mgrpw@localhost/")
+  >>> proxy.add(20, 22)
+  42
+
+Faults
+======
+
+If you need to raise an error, the prefered way to do it is via an
+`xmlrpclib.Fault`:
+
+  >>> try:
+  ...    import xmlrpclib
+  ... except ImportError:
+  ...    import xmlrpc.client as xmlrpclib
+
+  >>> class FaultDemo:
+  ...     def __init__(self, context, request):
+  ...         self.context = context
+  ...         self.request = request
+  ...
+  ...     def your_fault(self):
+  ...         return xmlrpclib.Fault(42, "It's your fault!")
+
+Now we'll register it as a view:
+
+  >>> from zope.configuration import xmlconfig
+  >>> ignored = xmlconfig.string("""
+  ... <configure
+  ...     xmlns="http://namespaces.zope.org/zope"
+  ...     xmlns:xmlrpc="http://namespaces.zope.org/xmlrpc"
+  ...     >
+  ...   <!-- We only need to do this include in this example,
+  ...        Normally the include has already been done for us. -->
+  ...   <include package="zope.app.publisher.xmlrpc" file="meta.zcml" />
+  ...
+  ...   <xmlrpc:view
+  ...       for="zope.site.interfaces.IFolder"
+  ...       methods="your_fault"
+  ...       class="zope.app.testing.xmlrpc.README.FaultDemo"
+  ...       permission="zope.ManageContent"
+  ...       />
+  ... </configure>
+  ... """)
+
+Now, when we call it, we get a proper XML-RPC fault:
+
+  >>> proxy = ServerProxy("http://mgr:mgrpw@localhost/")
+  >>> proxy.your_fault()
+  Traceback (most recent call last):
+  xmlrpc.client.Fault: <Fault 42: "It's your fault!">
+
+
+DateTime values
+===============
+
+Unfortunately, `xmlrpclib` does not support Python 2.3's new
+`datetime.datetime` class (it should be made to, really).  DateTime
```

Comment:
I see `import datetime` in `xmlrpclib` from Python 2.7.  This might be outdated.